### PR TITLE
Nr 164384 add macos roles

### PR DESCRIPTION
**Pull request by rogercoll**

### CODE REVIEW

File: caos.ansible_roles/roles/infra_agent/tasks/host/host-install-Darwin.yaml [hunk 1]
```diff
@@ -34,9 +39,14 @@
 
 - name: Install agent
   become: false
-  command: "{{ brew_path }}/bin/brew reinstall --formula --build-from-source /Users/{{ ansible_user }}/newrelic-infra-agent.rb"
+  command: "{{ brew_path }}/bin/brew reinstall --force --formula --build-from-source /Users/{{ ansible_user }}/newrelic-infra-agent.rb"
+
+# Need to download https://formulae.brew.sh/api/formula.jws.json but cannot as root! Run `brew update` without `sudo` first then try again.
```
Comment (rubenruizdegauna) [inline caos.ansible_roles/roles/infra_agent/tasks/host/host-install-Darwin.yaml:44]: Why is this 👇 being executed again?

File: caos.ansible_roles/roles/nr_repo_setup/tasks/main.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 ---
 
 - ansible.builtin.include_tasks: "repo-{{ ansible_distribution }}.yaml"
-  when: ansible_distribution != "MacOSX"
-
-- name:
-  debug:
-    msg: "repo-MacOSX.yaml is not refactored to be generic yet"
-  when: ansible_distribution == "MacOSX"
+#   when: ansible_distribution != "MacOSX"
```
Comment (rubenruizdegauna) [inline caos.ansible_roles/roles/nr_repo_setup/tasks/main.yaml:4]: As this is being fixed I would remove the comment :)

Comment (rogercoll) [reply caos.ansible_roles/roles/nr_repo_setup/tasks/main.yaml]: Good catch, removed in https://github.com/newrelic-experimental/caos-ansible-roles/pull/68/commits/bcc8d782da8e841d3b65fdfc0fa464e08fe7c5e4

File: caos.ansible_roles/roles/infra_agent/tasks/host/host-install-Darwin.yaml (hunk 1 above)
Comment (rogercoll) [reply caos.ansible_roles/roles/infra_agent/tasks/host/host-install-Darwin.yaml]: I think after installing, the infra-agent was showing the "update" message, but most probably a leftover. Let me remove it: https://github.com/newrelic-experimental/caos-ansible-roles/pull/68/commits/b197b61fcb352779c086dad9c9795a37b9aedb4e

**Pull request merged**